### ISSUE TITLE
Feature matching loss computation

### ISSUE THREAD

**Issue by LucaLaFisca** [xiangzhang1015/Adversarial-Variational-Semi-supervised-Learning#4]
The way you compute your feature matching loss at line 169 in AVAE.py seems unusual:
loss_featurematch = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(labels=d_data, logits=middle_feature))

It looks like you train your CNN to become an identity function when using this loss. Wouldn't be more efficient to compute the loss between the middle_feature corresponding to Zi with the one corresponding to Zs? Somehting like:
loss_featurematch = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(labels=middle_feature[0 : d_data.shape[0]//2, :], logits=middle_feature[d_data.shape[0]//2 :  -1, :]))
